src/librustdoc/html/render.rs

```diff
@@ -549,50 +588,72 @@ fn build_index(krate: &clean::Crate, cache: &mut Cache) -> io::Result<String> {
+             &cx.dst.join("main.js"));
+    try_err!(write(cx.dst.join("playpen.js"),
+               include_bytes!("static/playpen.js")),
+             &cx.dst.join("playpen.js"));
+    try_err!(write(cx.dst.join("main.css"),
+               include_bytes!("static/main.css")),
+             &cx.dst.join("main.css"));
+    try_err!(write(cx.dst.join("normalize.css"),
+               include_bytes!("static/normalize.css")),
+             &cx.dst.join("normalize.css"));
+    try_err!(write(cx.dst.join("FiraSans-Regular.woff"),
+               include_bytes!("static/FiraSans-Regular.woff")),
+             &cx.dst.join("FiraSans-Regular.woff"));
+    try_err!(write(cx.dst.join("FiraSans-Medium.woff"),
+               include_bytes!("static/FiraSans-Medium.woff")),
+             &cx.dst.join("FiraSans-Medium.woff"));
+    try_err!(write(cx.dst.join("FiraSans-LICENSE.txt"),
+               include_bytes!("static/FiraSans-LICENSE.txt")),
+             &cx.dst.join("FiraSans-LICENSE.txt"));
+    try_err!(write(cx.dst.join("Heuristica-Italic.woff"),
+               include_bytes!("static/Heuristica-Italic.woff")),
+             &cx.dst.join("Heuristica-Italic.woff"));
+    try_err!(write(cx.dst.join("Heuristica-LICENSE.txt"),
+               include_bytes!("static/Heuristica-LICENSE.txt")),
+             &cx.dst.join("Heuristica-LICENSE.txt"));
+    try_err!(write(cx.dst.join("SourceSerifPro-Regular.woff"),
+               include_bytes!("static/SourceSerifPro-Regular.woff")),
+             &cx.dst.join("SourceSerifPro-Regular.woff"));
+    try_err!(write(cx.dst.join("SourceSerifPro-Bold.woff"),
+               include_bytes!("static/SourceSerifPro-Bold.woff")),
+             &cx.dst.join("SourceSerifPro-Bold.woff"));
+    try_err!(write(cx.dst.join("SourceSerifPro-LICENSE.txt"),
+               include_bytes!("static/SourceSerifPro-LICENSE.txt")),
+             &cx.dst.join("SourceSerifPro-LICENSE.txt"));
+    try_err!(write(cx.dst.join("SourceCodePro-Regular.woff"),
+               include_bytes!("static/SourceCodePro-Regular.woff")),
+             &cx.dst.join("SourceCodePro-Regular.woff"));
+    try_err!(write(cx.dst.join("SourceCodePro-Semibold.woff"),
+               include_bytes!("static/SourceCodePro-Semibold.woff")),
+             &cx.dst.join("SourceCodePro-Semibold.woff"));
+    try_err!(write(cx.dst.join("SourceCodePro-LICENSE.txt"),
+               include_bytes!("static/SourceCodePro-LICENSE.txt")),
+             &cx.dst.join("SourceCodePro-LICENSE.txt"));
+    try_err!(write(cx.dst.join("LICENSE-MIT.txt"),
+               include_bytes!("static/LICENSE-MIT.txt")),
+             &cx.dst.join("LICENSE-MIT.txt"));
+    try_err!(write(cx.dst.join("LICENSE-APACHE.txt"),
+               include_bytes!("static/LICENSE-APACHE.txt")),
+             &cx.dst.join("LICENSE-APACHE.txt"));
+    try_err!(write(cx.dst.join("COPYRIGHT.txt"),
+               include_bytes!("static/COPYRIGHT.txt")),
+             &cx.dst.join("COPYRIGHT.txt"));
```

Comment:
I think that this can be a lot less wordy by returning the appropriate error in the `write` function rather than adding the filename here.